bbin: init at 0.1.4

Sohalt opened the pull request:
###### Description of changes


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Sohalt committed:
bbin: init at 0.1.4

AndersonTorres commented on pkgs/development/tools/bbin/default.nix:
```diff
@@ -0,0 +1,37 @@
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  dontBuild = true;
```
`dontConfigure` too:


```suggestion
  dontConfigure = true;
  dontBuild = true;
```

AndersonTorres commented on pkgs/development/tools/bbin/default.nix:
```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
```
```suggestion
, stdenvNoCC
```

because no C compilation occurs here.

AndersonTorres commented on pkgs/development/tools/bbin/default.nix:
```diff
@@ -0,0 +1,37 @@
+
+  dontBuild = true;
+
+  installPhase = ''
```
Decoration:
```suggestion
  installPhase = ''
    runHook preInstall

```

AndersonTorres commented on pkgs/development/tools/bbin/default.nix:
```diff
@@ -0,0 +1,37 @@
+    install -D bbin $out/bin/bbin
+    wrapProgram $out/bin/bbin \
+      --prefix PATH : "${lib.makeBinPath [ babashka jdk ]}"
+  '';
```
```suggestion

    runHook postInstall
  '';
```

AndersonTorres commented on pkgs/development/tools/bbin/default.nix:
```diff
@@ -0,0 +1,37 @@
+  dontBuild = true;
+
+  installPhase = ''
+    install -D bbin $out/bin/bbin
```
Can you install docs too?

AndersonTorres commented on pkgs/development/tools/bbin/default.nix:
```diff
@@ -0,0 +1,44 @@
+    homepage = "https://github.com/babashka/bbin";
+    description = "Install any Babashka script or project with one command";
+    license = licenses.mit;
+    platforms = babashka.meta.platforms;
```
Shorter and more elegant:

```suggestion
    inherit (babashka.meta) platforms;
```

Sohalt replied:
It's definitely shorter, but I'd argue about more elegant.
The other way the attrnames are easier to see, imho.

AndersonTorres replied:
This is a very common pattern in Nixpkgs:

```
$ grep -Iirn "inherit" pkgs/ 
15841
```

AndersonTorres reviewed (changes requested):
Otherwise lgtm

Sohalt commented:
Thanks @AndersonTorres for the review! I addressed all your points.
However I currently get
```
24:   ["bb" "--deps-root" script-root "--config" (str tmp-edn)
25:         script-main-opts-first script-main-opts-second
26:         "--"])
27: 
28: (process/exec (into base-command *command-line-args*))
    ^--- exec is not supported in non-GraalVM environments
```
when trying to execute a script installed with bbin.
Not entirely sure what's going on there, since babashka is using GraalVM.
Maybe some of the babashka maintainers know more? cc @bandresen @bhougland @DerGuteMoritz @jlesquembre @thiagokokada

Sohalt commented:
Ok, the issue is that the check for whether `process/exec` is supported, happens at compile time and nixpkgs uses the babashka-standalone.jar from the babashka-releases, which used to be built with AdoptOpenJDK (it's built with GraalVM since https://github.com/babashka/babashka/commit/a5dec07279184355a7de0813ec19401667c75a14). So this should work without changes to this PR as soon as the next version gets released and packaged for nixpkgs.

AndersonTorres commented:
So you need a new babashka in order to your package works?

Sohalt commented:
Exactly.

Sohalt commented:
Should work now with the newer babashka in nixpkgs.
